### PR TITLE
[build-script] Isolate CMake from the system

### PR DESCRIPTION
**Pull request by DougGregor**
When building on Darwin, we're supposed to build against the SDK,
not the host system. Isolate the build further from the system.

This continues the work of https://github.com/apple/swift/pull/32419

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci test

**Comment by DougGregor** [comment]
@swift-ci please test source compatibility

**Review by edymtt — APPROVED** [review]

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

File: utils/build-script [hunk 1]
```diff
@@ -701,6 +701,11 @@ class BuildScriptInvocation(object):
                 "--host-lipo", toolchain.lipo,
             ]
 
+            # Isolate build from the system; Darwin toolchains build against SDKs.
+            args.extra_cmake_options.append(
+                '-DCMAKE_IGNORE_PATH=/usr/lib;/usr/local/lib;/lib'
```
Comment (gottesmm) [inline utils/build-script:706]: Can you add to the comment here what type of problems result from us not doing this?

Comment (gottesmm) [reply utils/build-script:706]: Or you know what. I think this is pretty common sense. Nm.

**Pull request merged**